### PR TITLE
mungegithub: publish apiextensions-apiserver

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by deads2k** [comment]
It's a thing I want.  I don't know enough about the code to intelligently review it.

**Comment by caesarxuchao** [comment]
The code looks fine. Have you tried to run it locally? Or i can do it as part of solving https://github.com/kubernetes/test-infra/issues/2847.

**Comment by caesarxuchao** [comment]
@deads2k can you create the api-extensions repo?

**Comment by deads2k** [comment]
> @deads2k can you create the api-extensions repo?

I cannot.  @lavalamp has the power.

**Comment by lavalamp** [comment]
Where did we discuss the exact name? I am not a fan of "api-extensions"--it will confuse people.

**Comment by deads2k** [comment]
The name matches the group, so here https://github.com/kubernetes/kubernetes/issues/46702, here https://github.com/kubernetes/kubernetes/pull/46775, here https://github.com/kubernetes/kubernetes/pull/45115,  here https://github.com/kubernetes/kubernetes/issues/45277, and here https://github.com/kubernetes/kubernetes/pull/45822

**Comment by lavalamp** [comment]
I'm very sorry I didn't notice before, I thought it was called the custom resource apiserver since everyone refers to it as CRD. I don't think apiextensions is a good choice for the name, neither for the group nor the binary.

**Comment by sttts** [comment]
> Have you tried to run it locally? Or i can do it as part of solving #2847.

Haven't tried locally (would have to find out how). So with #2847 is fine.

**Comment by sttts** [comment]
@caesarxuchao https://github.com/kubernetes/apiextensions-apiserver exists now. Feel free to test with #2847. Let me know if you need any help or review.

**Comment by eparis** [comment]
Although now that I created the repo I see that @lavalamp doesn't like the name. So lets wait until he comments to decide if I should destroy it or not...

Daniel?

**Comment by lavalamp** [comment]
I think I've been persuaded to go with that name.

**Comment by caesarxuchao** [comment]
@sttts will do it today.

**Comment by sttts** [comment]
> will do it today.

Cool, thanks!

**Comment by caesarxuchao** [comment]
@sttts @deads2k could you manually cherry-pick one or two initial [PRs](https://github.com/kubernetes/kubernetes/commits/master/staging/src/k8s.io/apiextensions-apiserver) to the apiextensions repo? Currently it's empty and i can't fork it.

**Comment by deads2k** [comment]
> @sttts @deads2k could you manually cherry-pick one or two initial PRs to the apiextensions repo? Currently it's empty and i can't fork it.

Updated with a readme and license so that it won't conflict with any initial commits: https://github.com/kubernetes/apiextensions-apiserver

**Comment by caesarxuchao** [comment]
There are a few bugs in the PR. I'll send a PR once local tests pass.

**Comment by caesarxuchao** [comment]
I'm merging this PR as-is.

File: mungegithub/mungers/publisher.go [hunk 1]
```diff
@@ -211,8 +211,25 @@ func (p *PublisherMunger) Initialize(config *github.Config, features *features.F
 		publishScript: "/publish_scripts/publish_sample_apiserver.sh",
 	}
 
+	apiExtensionsAPIServer := repoRules{
+		dstRepo: "apiextensions-apiserver",
+		srcToDst: []branchRule{
+			{
+				// rule for the sample-apiserver master branch
+				src: coordinate{repo: config.Project, branch: "master", dir: "staging/src/k8s.io/apiextensions-apiserver"},
+				dst: coordinate{repo: "sample-apiserver", branch: "master", dir: "./"},
```
Comment (sttts) [inline mungegithub/mungers/publisher.go]: wrong repo

**Pull request merged**